tools/android/json_worker_wrapper.py

```diff
@@ -0,0 +1,32 @@
+import json
+import os
+import sys
+import traceback
+
+
+def wrap_worker(flags, main, app_run):
+  if sys.argv[1] == "--persistent_worker":
+    while True:
+      # parse work requests from stdin
+      request_str = sys.stdin.readline()
+      work_request = json.loads(request_str)
+
+      args = [sys.argv[0]] + work_request["arguments"]
+      flags(args)
+      _wrap_result(main, app_run, args)
+  else:
+    flags(sys.argv)
+    app_run(main)
+
+
+def _wrap_result(main, app_run, args):
+  response = {
+    "exitCode": 0,
+    "output": "",
+  }
+  try:
+    app_run(main, argv=args, exit=False)
+  except Exception as e:
+    response["exitCode"] = 1
+    response["output"] = "Exception occurred: {}\nTraceback:\n{}".format(e, traceback.format_exc())
+  os.write(1, json.dumps(response).encode("utf-8"))
```

Comment:
Hard-coding `1` for stdin here seems like a bad idea. Not sure if this works outside of Linux, not to mention that it is hard to understand. If there are any python convenience macros (maybe `sys.stdin`?), that would be preferable.